network/p2p/libp2pNode.go

```diff
@@ -79,17 +46,93 @@ type LibP2PNode interface {
 	PeerManagerComponent() component.Component
 	// RequestPeerUpdate requests an update to the peer connections of this node using the peer manager.
 	RequestPeerUpdate()
+}
+
+// Routable set of node routing capabilities
+type Routable interface {
+	// RoutingTable returns the node routing table
+	RoutingTable() *kbucket.RoutingTable
 	// SetRouting sets the node's routing implementation.
 	// SetRouting may be called at most once.
 	SetRouting(r routing.Routing)
 	// Routing returns node routing object.
 	Routing() routing.Routing
+}
+
+// PeerToPeer peers list operations
+type PeerToPeer interface {
+	// AddPeer adds a peer to this node by adding it to this node's peerstore and connecting to it.
+	AddPeer(ctx context.Context, peerInfo peer.AddrInfo) error
+	// RemovePeer closes the connection with the peer.
+	RemovePeer(peerID peer.ID) error
+	// ListPeers returns list of peer IDs for peers subscribed to the topic.
+	ListPeers(topic string) []peer.ID
+	// GetPeersForProtocol returns slice peer IDs for the specified protocol ID.
+	GetPeersForProtocol(pid protocol.ID) peer.IDSlice
+}
+
+// StreamManagement peer to peer stream management functions
+type StreamManagement interface {
+	// CreateStream returns an existing stream connected to the peer if it exists, or creates a new stream with it.
+	CreateStream(ctx context.Context, peerID peer.ID) (libp2pnet.Stream, error)
+}
+
+// PubSub publish subscribe features for node
+type PubSub interface {
+	// Subscribe subscribes the node to the given topic and returns the subscription
+	Subscribe(topic channels.Topic, topicValidator TopicValidatorFunc) (Subscription, error)
+	// UnSubscribe cancels the subscriber and closes the topic.
+	UnSubscribe(topic channels.Topic) error
+	// Publish publishes the given payload on the topic.
+	Publish(ctx context.Context, topic channels.Topic, data []byte) error
 	// SetPubSub sets the node's pubsub implementation.
 	// SetPubSub may be called at most once.
 	SetPubSub(ps PubSubAdapter)
-	// SetComponentManager sets the component manager for the node.
-	// SetComponentManager may be called at most once.
-	SetComponentManager(cm *component.ComponentManager)
+}
+
+// LibP2PNode represents a flow libp2p node. It provides the network layer with the necessary interface to
```

Comment:
`flow` is recommended to be written as a noun, i.e., `Flow`.

Reply:
https://github.com/onflow/flow-go/pull/4630/commits/455147a08f62bace4ff3fb18c2cc6ae018d3ca68